pandas/tests/dtypes/test_cast.py

```diff
@@ -456,3 +457,17 @@ def test_cast_1d_arraylike_from_scalar_categorical(self):
 def test_construct_1d_ndarray_preserving_na(values, dtype, expected):
     result = construct_1d_ndarray_preserving_na(values, dtype=dtype)
     tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.mark.parametrize('arr, dtype, expected', [
+    (np.array(['0:0:1'], dtype='object'),
+     'timedelta64[ns]', 'timedelta64[ns]'),
+    (np.array(['2000'], dtype='object'),
+     'datetime64[ns]', 'datetime64[ns]'),
+    (np.array(['2000'], dtype='object'),
+     'datetime64', 'datetime64[ns]'),
```

Comment:
These can be made less verbose (easier to read if on one line, but not a huge deal) by changing `'object'` --> `'O'`, `'timedelta64[ns]'` --> `'m8[ns]'`, and `'datetime64[ns]''` --> `M8[ns]`